src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala

```diff
@@ -286,6 +284,12 @@ object ErgoStateContext {
 case class ErgoStateContextSerializer(votingSettings: VotingSettings) extends ScorexSerializer[ErgoStateContext] {
 
   override def serialize(obj: ErgoStateContext, w: Writer): Unit = {
+    /* NOHF PROOF:
+    Changed: added assert to not let `UpcomingStateContext` get serialized.
+    Motivation: only `ErgoStateContext` is supported in `parse`.
+    Safety: used only in `ErgoMiner.createCandidate` and does get serialized.
```

Comment:
not clear what *used only in ...*?

Reply:
Yes, somehow I missed the main part. :) Fixed to "Safety: `UpcomingStateContext` is used only in `ErgoMiner.createCandidate` and does not get serialized."